docs: add tabs for scala 3 (#1201)
zio/zio-json: README.md

into a Scala `case class`

```scala
final case class Banana(curvature: Double)
```

To do this, we derive an *instance* of the `JsonDecoder` typeclass for `Banana`.

```scala
final case class Banana(curvature: Double) derives JsonDecoder
```

> [!NOTE]
>

> > In scala 2, we need to use the `zio-json` code generator. It is best practice to put it on the companion of `Banana`, like so

Reviewer: "code generator" feels like there's an external tool to use to make it generate some code

```suggestion
> In scala 2, we need to use the `zio-json` semi-automatic derivation. It is best practice to put it on the companion of `Banana`, like so
```

> case class Banana(curvature: Double)

Reviewer: ```suggestion
final case class Banana(curvature: Double)
```

```

To do this, we derive an *instance* of the `JsonDecoder` typeclass for `Banana`.

```scala

> case class Banana(curvature: Double) derives JsonDecoder

Reviewer: ```suggestion
final case class Banana(curvature: Double) derives JsonDecoder
```

```

> [!NOTE]
>
> In scala 2, we need to use the `zio-json` code generator. It is best practice to put it on the companion of `Banana`, like so
>
> ```scala
> object Banana {
>   implicit val decoder: JsonDecoder[Banana] = DeriveJsonDecoder.gen[Banana]
> }
> ```

Now we can parse JSON into our object

```
scala> """{"curvature":0.5}""".fromJson[Banana]
val res: Either[String, Banana] = Right(Banana(0.5))
```

Likewise, to produce JSON from our data we derive a `JsonEncoder`

```scala

> case class Banana(curvature: Double) derives JsonEncoder

Reviewer: ```suggestion
final case class Banana(curvature: Double) derives JsonEncoder
```

Say we extend our data model to include more data types

```scala
enum Fruit:
  case Banana(curvature: Double) extends Fruit

>   case Apple(poison: Boolean)    extends Fruit

Reviewer: ```suggestion
enum Fruit {
  case Banana(curvature: Double)
  case Apple(poison: Boolean)
}
```

```

we can generate the encoder and decoder for the entire `sealed` family using `JsonCodec`

```scala

> enum Fruit derives JsonCodec:
>   case Banana(curvature: Double)
>   case Apple(poison: Boolean)

Reviewer: ```suggestion
enum Fruit derives JsonCodec {
  case Banana(curvature: Double)
  case Apple(poison: Boolean)
}
```

```

> [!NOTE]
>
> In scala 2:
>
> ```scala mdoc:compile-only
> import zio.json._
> 

> > sealed trait Fruit                   extends Product with Serializable

Reviewer: ```suggestion
> sealed trait Fruit
```

> case class Banana(curvature: Double) extends Fruit

> > case class Apple(poison: Boolean)    extends Fruit

Reviewer: ```suggestion
> final case class Banana(curvature: Double) extends Fruit
> final case class Apple(poison: Boolean)    extends Fruit
```
